rawtodigi_alpaka.h

```diff
@@ -2,19 +2,25 @@
 #define rawtodigi_alpaka_h_
 
 #include "alpakaConfig.h"
-#include "input.h"
-#include "output.h"
-#include "pixelgpudetails.h"
+
+class Input;
+class Output;
 
 namespace ALPAKA_ARCHITECTURE {
 
-  void rawtodigi(const Input *input_d,
-                 Output *output_d,
-                 const uint32_t wordCounter,
-                 bool useQualityInfo,
-                 bool includeErrors,
-                 bool debug,
-                 Queue queue);
+  struct rawtodigi_kernel {
+    template <typename T_Acc>
+    ALPAKA_FN_ACC void operator()(T_Acc const& acc,
+                                  const Input* input,
+                                  Output* output,
+                                  bool useQualityInfo,
+                                  bool includeErrors,
+                                  bool debug) const;
+  };
+
+  // explicit template instantiation declaration for ALPAKA_ARCHITECTURE::Acc
+  extern template ALPAKA_FN_ACC void rawtodigi_kernel::operator()(
+      Acc const& acc, const Input* input, Output* output, bool useQualityInfo, bool includeErrors, bool debug) const;
```

Comment:
So the trick is that the caller (`analyzer_alpaka.cc`) is compiled separately for each backend, as is the implementation file (`rawtodigi_alpaka.cc`), and in `rawtodigi_alpaka.cc` the `ALPAKA_ARCHITECTURE::Acc` type drives the instantiation. The setup works as long as we compile the caller and the implementation for the same backend, and link them together. Did I get it right?

Do you have any other reason in mind for this separation of caller and implementation (i.e. not #including the implementation into the caller) than the hope for compiling the caller with the host compiler?

Reply:
> Did I get it right?

Correct.

> Do you have any other reason in mind for this separation of caller and implementation (i.e. not #including the implementation into the caller) than the hope for compiling the caller with the host compiler?

No, that is the main reason, apart from the usual reason of trying to keep the individual files to manageable size.